docs/advanced_topics/icons.md

```diff
@@ -30,7 +30,7 @@ Draw or download an icon and save it in a template folder:
 
 The `svg` tag should:
 
--   Set the `id="icon-<name>"` attribute, icons are referenced by this name.
+-   Set the `id="icon-<name>"` attribute, icons are referenced by this `name`. The `name` should only contain lowercase letters, numbers, and hyphens.
```

Comment:
Pretty sure this limitation only exists because of the styleguide's regex, and the icons can technically still be used elsewhere as long as the ID is prefixed with `icon-`. Alternatively we could update the regex to be more permissive.

Reply:
I think alphanumeric hyphenated should be fine? I’d be surprised if XML allowed much more.

Reply:
I think it allows at least a few others (e.g. `:`, `.`, `_`)
https://www.w3.org/TR/REC-xml/#NT-Name

But yeah I don't think we want to encourage people from using unconventional IDs as that may cause other unexpected issues down the line.